### PR TITLE
Me/record field temporal extents

### PR DESCRIPTION
**Pull request by LHBruneton-C2C**
### Description

This PR introduces:
- the sortable list element
- the date picker input

The date range picker input has been made fully functional.

Both date picker and date range picker are full width, to adapt to the parent design.

The record form field now supports temporal extents.

### Screenshots

image

### Quality Assurance Checklist

- [X] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [ ] If **new logic** ⚙️ is introduced: unit tests were added
- [ ] If **new user stories** 🤏 are introduced: E2E tests were added
- [X] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

### CODE REVIEW

**Comment by LHBruneton-C2C** [comment]
@jahow Thanks for the pre-review, and the template suggestion. I don't see how to make it work with user input though. Here, in the table, you're just displaying data. But in the Sortable list, the elements can be added/reordered/deleted, but their content can be changed too. That's where I found the ComponentOutlet most useful, as it can take an input (and an output, by making the input subscribable).

**Comment by LHBruneton-C2C** [comment]
@jahow If you didn't before, you can do a full review now. If you already did, I only added some changes with the date picker and date range picker design.

**Review by jahow — APPROVED** [review]
Thank you!! It works well in the editor form. I made a few comments and I think the date and range pickers will still need some work (e.g. the range picker closing after first click, the pickers not showing up when clicking the text, the date format not being localized).

I also feel that the form-field-temporal-extents component ends up being quite complex, either Angular is really making this complicated or there is a simpler way to have the same result that we're missing. Either way, I think we can keep it as it is for now :) and it's definitely not a trivial component anyways.

File: libs/feature/editor/src/lib/components/record-form/form-field/form-field-temporal-extents/form-field-temporal-extents.component.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  }>
+
+  addOptions$ = combineLatest([
+    this.translateService
+      .get('editor.record.form.temporalExtents.addDate')
+      .pipe(map((buttonLabel) => ({ buttonLabel, eventName: 'date' }))),
+    this.translateService
+      .get('editor.record.form.temporalExtents.addRange')
+      .pipe(map((buttonLabel) => ({ buttonLabel, eventName: 'range' }))),
+  ])
```
[the remark this answers is not in the document]
Comment (LHBruneton-C2C) [reply libs/feature/editor/src/lib/components/record-form/form-field/form-field-temporal-extents/form-field-temporal-extents.component.ts:43]: I kinda feel that the component responsible for deleting elements should also deal with re-adding them...

**Pull request merged**